### PR TITLE
OpenShiftOAuthInterceptor will now try with oauth token, if original …

### PR DESCRIPTION
**Pull request by iocanel**
…request is unauthorized and request using credentials is also unauthorized.

### CODE REVIEW

**Comment by jimmidyson** [comment]
Can you explain a bit more about what the use case is for this? I assume that if a user has decided to use the username/password then that should be the only auth used, not fall back to a token.

**Comment by iocanel** [comment]
@jimmidyson: That would be the case if username / password were only set explicitly, but this is not the case. A username/password may have been picked up from system / environment / .kube/config and will hide oauth regardless of how it was set (even if it has been set explicitly).

**Comment by jimmidyson** [comment]
I would say that is fine behaviour though. If the env has stuff that config is set from, then that should be respected, not fall back to oauth token. Personally I would find that surprising behaviour.

**Comment by iocanel** [comment]
You can tell that to the person that writes the code bellow and gets an Exception.

    Config config = new ConfigBuilder().withMasterUrl(master).withOauthToken(token).build();
    OpenshiftClient client = new DefaultOpenshiftClient(config);
    client.doStuff();

**Comment by jimmidyson** [comment]
Now that is different. We should probably fix the fact that if any auth stuff is explicitly set through builder then that takes precedence over discovered from environment.

File: openshift-client/src/main/java/io/fabric8/openshift/client/internal/OpenShiftOAuthInterceptor.java [hunk 1]
```diff
@@ -122,5 +109,37 @@ private  String authorize() {
...
+  }
+
+  private static void setAuthHeader(Request.Builder builder, String token) {
+    if (token != null) {
+      builder.header(AUTHORIZATION, String.format("Bearer %s", token));
+    }
+  }
+
+  private static boolean isUnauthorized(Response response) {
+    return response.code() == 401 || response.code() == 403;
```
Comment (jimmidyson) [inline openshift-client/src/main/java/io/fabric8/openshift/client/internal/OpenShiftOAuthInterceptor.java]: Please use `HttpURLConnection.HTTP_OK` and `HttpURLConnection.HTTP_FORBIDDEN` just to be a bit clearer what these mean.

**Comment by jimmidyson** [comment]
Minor nit but as discussed on IRC earlier, let's merge this before next release.

**Pull request closed without merging**